support/ebpf/process_monitor.ebpf.c

```diff
@@ -0,0 +1,26 @@
+// This file contains the code for the kprobe that reports the exit of a thread group.
+
+#include "bpfdefs.h"
+#include "tracemgmt.h"
+
+#include "types.h"
+
+// Every time a thread group exits this hook is triggered.
+SEC("kprobe/disassociate_ctty")
+int disassociate_ctty(struct pt_regs *ctx)
```

Comment:
I think this needs a better research / documentation:
 - kprobe is previously unneeded feature iirc.
 - tracepoint is faster than kprobe
 - `disassociate_ctty` is not available if `CONFIG_TTY=n`
 - the code here is wrong, `disassociate_ctty(0)` is called when tty is removed without exit, it is called with argument 1 if process is existing
 - is the function signature consistent throughout supported kernel versions?
 - given that this is not an exported function and has only two call sites with different argument, the compiler might make specialized copies of it depending on optimization and lto settings